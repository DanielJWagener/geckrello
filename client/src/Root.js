// Simply creates a Redux store and passes it in to a provider tag.
// This logic was originally in index.js, but we've movied it here to
// make our testing process easier.
import React from "react";
import { Provider } from "react-redux";
import { createStore, applyMiddleware, compose } from "redux";
import reduxThunk from "redux-thunk";
import reducers from "./reducers";

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(
  reducers,
  {},
  composeEnhancers(applyMiddleware(reduxThunk))
);

export default ({ children }) => {
  return <Provider store={store}>{children}</Provider>;
};
