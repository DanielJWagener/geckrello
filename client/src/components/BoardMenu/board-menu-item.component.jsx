import React from "react";

const BoardMenuItem = ({ togglePanel, toggleTarget, children }) => (
  <button className="boardmenu__item" onClick={() => togglePanel(toggleTarget)}>
    {children}
  </button>
);

export default BoardMenuItem;
