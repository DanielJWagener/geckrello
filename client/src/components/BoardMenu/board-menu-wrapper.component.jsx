import React from "react";
import ReactDOM from "react-dom";

import "./board-menu-wrapper.styles.scss";

class BoardMenuWrapper extends React.Component {
  setVh = () => {
    let vh = window.innerHeight * 0.01;
    document.documentElement.style.setProperty("--vh", `${vh}px`);
  };

  componentDidMount() {
    this.setVh();
    window.addEventListener("resize", this.setVh);
  }

  render() {
    return ReactDOM.createPortal(
      <div className="sidebar__wrapper">
        <div
          className={`sidebar sidebar--${
            this.props.menuHidden ? "hidden" : "shown"
          }`}
        >
          {this.props.children}
        </div>
      </div>,
      document.getElementById("sidebar-root")
    );
  }
}

export default BoardMenuWrapper;
