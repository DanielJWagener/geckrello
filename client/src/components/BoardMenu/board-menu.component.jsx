import React from "react";
import BoardMenuContent from "../BoardMenuContent/board-menu-content.component";

import "./board-menu.styles.scss";

const sidebarRoot = document.getElementById("sidebar-root");

class BoardMenu extends React.Component {
  constructor(props) {
    super(props);
    this.el = document.createElement("div");
  }

  setVh = () => {
    let vh = window.innerHeight * 0.01;
    document.documentElement.style.setProperty("--vh", `${vh}px`);
  };

  componentDidMount() {
    this.setVh();
    window.addEventListener("resize", this.setVh);
    sidebarRoot.appendChild(this.el);
  }

  componentWillUnmout() {
    sidebarRoot.removeChild(this.el);
  }

  render() {
    return (
      <div className="sidebar__wrapper">
        <div
          className={`sidebar sidebar--${
            this.props.menuHidden ? "hidden" : "shown"
          }`}
        >
          <BoardMenuContent toggleMenu={this.props.toggleMenu} />
        </div>
      </div>
    );
  }
}

export default BoardMenu;
