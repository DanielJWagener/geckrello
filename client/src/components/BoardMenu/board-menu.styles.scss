@import "../../sass/abstracts/_variables.scss";

.portal-inner {
  height: 100%;
}

.sidebar {
  position: absolute;

  &--hidden {
    left: 0rem;
  }

  &--shown {
    left: calc(-1 * #{$sidebar-width});
  }

  height: calc(var(--vh, 1vh) * 100 - 9rem);
  background-color: $component-background;
  border-left: 2px solid white;
  border-radius: 8px 0 0 8px;
  width: $sidebar-width;
  padding: 1.5rem;
  transition: left 0.3s;

  @media only screen and (max-width: 400px) {
    width: 100vw;
  }

  &__wrapper {
    position: absolute;
    top: 0;
    left: 0;
    height: 100%;
    width: $sidebar-width;
  }

  .sidebar-menu {
    &__heading {
      font-size: 2rem;
      font-weight: 700;
      text-align: center;
      margin-bottom: 1.5rem;
    }

    &__close {
      position: absolute;
      top: 3px;
      right: 1rem;
      cursor: pointer;
      font-size: 3rem;

      &:hover {
        color: red;
      }
    }

    &__interface {
      position: absolute;
      top: 0;
      width: $sidebar-width;

      height: calc(var(--vh, 1vh) * 100 - 9rem);
      overflow-y: auto;
      background-color: white;
      border-left: 2px solid $component-background;
      border-radius: 8px 0 0 8px;
      padding: 1.5rem;
      transition: all 0.3s;

      @media only screen and (max-width: 400px) {
        width: 100vw;
      }

      &--hidden {
        right: calc(-1 * #{$sidebar-width});
      }

      &--visible {
        right: 0;
      }

      &--close {
        position: absolute;
        top: 3px;
        right: 1rem;
        cursor: pointer;
        font-size: 3rem;

        &:hover {
          color: red;
        }
      }
    }
  }
}

.boardmenu__item {
  font-size: 2rem;
  padding: 2rem 1rem 2rem 1rem;
  width: 100%;
  text-align: left;
  border: none;
  border-radius: 3px;
  background-color: $component-background;
  cursor: pointer;
  &:hover {
    background-color: $component-background-dark;
  }
}

.boardmenu__panel {
  position: absolute;
  top: 0;
  width: $sidebar-width;

  height: calc(var(--vh, 1vh) * 100 - 9rem);
  overflow-y: auto;
  background-color: white;
  border-left: 2px solid $component-background;
  border-radius: 8px 0 0 8px;
  padding: 1.5rem;
  transition: all 0.3s;

  @media only screen and (max-width: 400px) {
    width: 100vw;
  }

  &--hidden {
    right: calc(-1 * #{$sidebar-width});
  }

  &--visible {
    right: 0;
  }

  &--close {
    position: absolute;
    top: 3px;
    right: 1rem;
    cursor: pointer;
    font-size: 3rem;

    &:hover {
      color: red;
    }
  }
}

.boardmenu__heading {
  font-size: 2rem;
  font-weight: 700;
  text-align: center;
  margin-bottom: 1.5rem;
}
