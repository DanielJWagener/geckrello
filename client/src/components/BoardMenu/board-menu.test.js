import { shallow } from "enzyme";
import React from "react";
import BoardMenu from "./board-menu.component";
import Root from "../../Root";

it("renders BoardMenu component", () => {
  expect(shallow(<BoardMenu menuHidden />).debug()).toMatchSnapshot();
});

it("is hidden on mount", () => {
  const wrapper = shallow(<BoardMenu menuHidden />);

  expect(wrapper.find(".sidebar--hidden").length).toEqual(1);
});
