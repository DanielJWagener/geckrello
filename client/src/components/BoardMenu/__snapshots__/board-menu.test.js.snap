// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders BoardMenu component 1`] = `
"<div className=\\"sidebar__wrapper\\">
  <div className=\\"sidebar sidebar--hidden\\">
    <BoardMenu toggleMenu={[undefined]} />
  </div>
</div>"
`;
